data/types_test.go

```diff
@@ -237,7 +238,7 @@ func TestDelegatedRoleUnmarshalErr(t *testing.T) {
 
 	// test for type errors
 	err := json.Unmarshal([]byte(`{"keyids":"a"}`), &d)
-	assert.Equal(t, "keyids", err.(*json.UnmarshalTypeError).Field)
+	assert.Equal(t, "json: slice unexpected end of JSON input", err.(*json.SyntaxError).Error())
```

Comment:
This is the only test that has been updated because it relies on the error message output of the json decoder